src/main/java/com/example/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public ShoppingCartResponseDto getShoppingCartByUserEmail(String email) {
+        User user = getUserByEmail(email);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseGet(() -> registerNewShoppingCart(user));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public CartItemResponseDto addCartItem(String email, CartItemRequestDto cartItemRequestDto) {
+        User user = getUserByEmail(email);
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseGet(() -> registerNewShoppingCart(user));
+        Optional<CartItem> existingCartItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId().equals(cartItemRequestDto.getBookId()))
+                .findFirst();
+        CartItem cartItem;
+        if (existingCartItem.isPresent()) {
+            cartItem = existingCartItem.get();
+            cartItem.setQuantity(cartItemRequestDto.getQuantity());
+        } else {
+            cartItem = createNewCartItem(cartItemRequestDto, shoppingCart);
+        }
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public CartItemResponseDto updateBookQuantity(Long cartItemId,
+                                                  CartItemQuantityDto cartItemQuantityDto) {
+        CartItem cartItem = cartItemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find cart items "
+                        + "by cart items id " + cartItemId));
+        cartItem.setQuantity(cartItemQuantityDto.getQuantity());
+        return cartItemMapper.toDto(cartItemRepository.save(cartItem));
+    }
+
+    @Override
+    public void deleteCartItemById(Long cartItemId) {
+        cartItemRepository.deleteById(cartItemId);
+    }
+
+    private User getUserByEmail(String email) {
+        return userRepository.findByEmail(email).orElseThrow(
+                () -> new EntityNotFoundException("Can not find user by email" + email));
+    }
+
+    private ShoppingCart registerNewShoppingCart(User user) {
```

Comment:
```suggestion
    public ShoppingCart registerNewShoppingCart(User user) {
```

call this method in register user method